api/lib/opentelemetry/distributed_context/distributed_context.rb

```diff
@@ -19,6 +19,8 @@ def entries
       def [](_key)
         nil
       end
+
+      def []=(_key, _value) end
```

Comment:
@mwear should this be part of the API? I wonder if we should gate this PR on #147 and build on the APIs defined there?

Reply:
@fbogsany since #147 is now merged I'll work on building off the APIs defined there tomorrow